storage/grpc_writer.go

```diff
@@ -818,12 +818,19 @@ func bidiWriteObjectRequest(r gRPCBidiWriteRequest, bufChecksum *uint32, objectC
 	return req
 }
 
-type getObjectChecksumsParams struct {
-	fullObjectChecksum  func() uint32
-	finishWrite         bool
+// checksumAttrs contains all parameters required from writer in senders,
+// helps avoid piping in writer to the individual senders.
+type checksumAttrs struct {
 	sendCRC32C          bool
 	disableAutoChecksum bool
-	attrs               *ObjectAttrs
+	objectAttrs         *ObjectAttrs
+	fullObjectChecksum  func() uint32
+}
+
+type getObjectChecksumsParams struct {
+	checksumAttrs
```

Comment:
Nested structs here seems confusing, why separate this out?

Reply:
Having getObjectChecksumParams as a type seems fine. It seems odd to use struct embedding here -- I have generally only seen it when e.g. we are implementing an interface or doing a union type. I also think checksumSettings might make more sense as a name than checksumAttrs? I see that you deleted the func that used to be called that.

Ultimately I don't think it's a huge deal, just found this a little confusing to read and follow.